ractor/src/pg/mod.rs

```diff
@@ -224,6 +412,37 @@ pub fn get_members(group_name: &GroupName) -> Vec<ActorCell> {
 ///
 /// Returns a [`Vec<GroupName>`] representing all the registered group names
 pub fn which_groups() -> Vec<GroupName> {
+    let monitor = get_monitor();
+    monitor
+        .map
+        .iter()
+        .map(|kvp| kvp.key().clone())
+        .map(|(_scope, group)| group.clone())
+        .collect::<Vec<_>>()
+}
+
+/// Returns a list of all known groups in scope `scope`
+///
+/// * `scope` - The scope to retrieve the groups from
+///
+/// Returns a [`Vec<GroupName>`] representing all the registered group names
+/// in `scope`
+pub fn which_groups_in_named_scope(scope: &ScopeName) -> Vec<GroupName> {
+    let monitor = get_monitor();
+    monitor
+        .map
+        .iter()
+        .map(|kvp| kvp.key().clone())
```

Comment:
This is where having another index might be helpful, i.e. Scope -> Group name in a separate mapping. But that's an optimization that can be added in the future.

Reply:
Left for the future :)